src/mobu/services/business/notebookrunner.py

```diff
@@ -272,14 +272,22 @@ async def execute_notebook(
     ) -> None:
         self._notebook = self.next_notebook()
         relative_notebook = self._relative_notebook()
+        logger = self.logger.bind(notebook=relative_notebook)
         msg = f"Notebook {self._notebook.name} iteration {iteration}"
-        self.logger.info(msg)
+        logger.info(msg)
 
         with self.trace_notebook(
             notebook=relative_notebook, iteration=iteration
         ) as span:
             try:
-                for cell in self.read_notebook(self._notebook):
+                cells = self.read_notebook(self._notebook)
+
+                # We want to wait if the notebook is totally empty so we don't
+                # spin out of control on empty notebooks
+                logger.warning("empty notebook")
+                if not cells:
+                    await self.execution_idle()
```

Comment:
This will probably get us to our load test goal without multiple replicas :smile: :cry: